pkg/hubble/relay/pool/client.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+)
+
+// ClientConn is an interface that defines the functions clients need to
+// perform unary and streaming RPCs. It is implemented by *grpc.ClientConn.
+type ClientConn interface {
+	// GetState returns the connectivity.State of ClientConn.
```

Comment:
Even in grpc v1.30.0 (the latest at the time of writing) `GetState()` is [marked as experimental](https://pkg.go.dev/google.golang.org/grpc?tab=doc#ClientConn.GetState).

Since this PR make use of it in several places, would it make sense to keep track of this somewhere?

Reply:
I think it's important to pay attention to this when upgrading the go-grpc dep but we will anyway have to face breaking changes when upgrading.